phpBB/phpbb/members/controller/team.php

```diff
@@ -0,0 +1,384 @@
+
+		// Can this user view profiles/memberlist?
+		if (!$this->auth->acl_gets('u_viewprofile', 'a_user', 'a_useradd', 'a_userdel'))
+		{
+			if ($this->user->data['user_id'] != ANONYMOUS)
+			{
+				throw new http_exception(403, 'NO_VIEW_USERS');
+			}
+
+			login_box('', $this->language->lang('LOGIN_EXPLAIN_TEAM'));
+		}
+
+		$sql = 'SELECT *
+			FROM ' . TEAMPAGE_TABLE . '
+			ORDER BY teampage_position ASC';
+		$result = $this->db->sql_query($sql, 3600);
+		$teampage_data = $this->db->sql_fetchrowset($result);
+		$this->db->sql_freeresult($result);
+
+		$sql_ary = array(
+			'SELECT'	=> 'g.group_id, g.group_name, g.group_colour, g.group_type, ug.user_id as ug_user_id, t.teampage_id',
+
+			'FROM'		=> array(GROUPS_TABLE => 'g'),
+
+			'LEFT_JOIN'	=> array(
+				array(
+					'FROM'	=> array(TEAMPAGE_TABLE => 't'),
+					'ON'	=> 't.group_id = g.group_id',
+				),
+				array(
+					'FROM'	=> array(USER_GROUP_TABLE => 'ug'),
+					'ON'	=> 'ug.group_id = g.group_id AND ug.user_pending = 0 AND ug.user_id = ' . (int) $this->user->data['user_id'],
+				),
+			),
+		);
+
+		$result = $this->db->sql_query($this->db->sql_build_query('SELECT', $sql_ary));
+
+		$group_ids = $groups_ary = array();
+		while ($row = $this->db->sql_fetchrow($result))
+		{
+			if ($row['group_type'] == GROUP_HIDDEN && !$this->auth->acl_gets('a_group', 'a_groupadd', 'a_groupdel') && $row['ug_user_id'] != $this->user->data['user_id'])
+			{
+				$row['group_name'] = $this->language->lang('GROUP_UNDISCLOSED');
+				$row['u_group'] = '';
+			}
+			else
+			{
+				$row['group_name'] = $this->group_helper->get_name($row['group_name']);
+				$row['u_group'] = append_sid("{$phpbb_root_path}memberlist.$phpEx", 'mode=group&amp;g=' . $row['group_id']);
+			}
+
+			if ($row['teampage_id'])
+			{
+				// Only put groups into the array we want to display.
+				// We are fetching all groups, to ensure we got all data for default groups.
+				$group_ids[] = (int) $row['group_id'];
+			}
+			$groups_ary[(int) $row['group_id']] = $row;
+		}
+		$this->db->sql_freeresult($result);
+
+		$sql_ary = array(
+			'SELECT'	=> 'u.user_id, u.group_id as default_group, u.username, u.username_clean, u.user_colour, u.user_type, u.user_rank, u.user_posts, u.user_allow_pm, g.group_id',
+
+			'FROM'		=> array(
+				USER_GROUP_TABLE => 'ug',
+			),
+
+			'LEFT_JOIN'	=> array(
+				array(
+					'FROM'	=> array(USERS_TABLE => 'u'),
+					'ON'	=> 'ug.user_id = u.user_id',
+				),
+				array(
+					'FROM'	=> array(GROUPS_TABLE => 'g'),
+					'ON'	=> 'ug.group_id = g.group_id',
+				),
+			),
+
+			'WHERE'		=> $this->db->sql_in_set('g.group_id', $group_ids, false, true),
+
+			'ORDER_BY'	=> 'u.username_clean ASC',
+		);
+
+		/**
+		 * Modify the query used to get the users for the team page
+		 *
+		 * @event core.memberlist_team_modify_query
+		 * @var array	sql_ary			Array containing the query
+		 * @var array	group_ids		Array of group ids
+		 * @var array	teampage_data	The teampage data
+		 * @since 3.1.3-RC1
+		 */
```

Comment:
Maybe we should add a `@changed` to inform developers this was moved to a controller?